packages/intrepid2/unit-test/Projection/test_interpolation_projection_TET.hpp

```diff
@@ -305,324 +329,328 @@ int InterpolationProjectionTet(const bool verbose) {
 
-          //compute Lagrangian Interpolation of fun
-          {
-            li::getDofCoordsAndCoeffs(dofCoordsOriented,  dofCoeffsPhys, basisPtr.get(), POINTTYPE_EQUISPACED, elemOrts);
+            //compute DofCoords Oriented
+            DynRankView ConstructWithLabel(dofCoordsOriented, numCells, basisCardinality, dim);
+            DynRankView ConstructWithLabel(dofCoeffsPhys, numCells, basisCardinality);
+            DynRankView ConstructWithLabel(physDofCoords, numCells, basisCardinality, dim);
+            DynRankView ConstructWithLabel(funAtDofCoords, numCells, basisCardinality);
+            DynRankView ConstructWithLabel(basisCoeffsLI, numCells, basisCardinality);
 
-            //Compute physical Dof Coordinates
+            //compute Lagrangian Interpolation of fun
             {
-              Basis_HGRAD_TET_C1_FEM<DeviceSpaceType,ValueType,ValueType> tetLinearBasis; //used for computing physical coordinates
-              DynRankView ConstructWithLabel(tetLinearBasisValuesAtDofCoords, numCells, tet.getNodeCount(), basisCardinality);
-              for(ordinal_type i=0; i<numCells; ++i)
-                for(ordinal_type d=0; d<dim; ++d) {
-                  auto inView = Kokkos::subview( dofCoordsOriented,i,Kokkos::ALL(),Kokkos::ALL());
-                  auto outView =Kokkos::subview( tetLinearBasisValuesAtDofCoords,i,Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL());
-                  tetLinearBasis.getValues(outView, inView);
-                  DeviceSpaceType().fence();
-                  for(ordinal_type j=0; j<basisCardinality; ++j)
-                    for(std::size_t k=0; k<tet.getNodeCount(); ++k)
-                      physDofCoords(i,j,d) += vertices[tets[i][k]][d]*tetLinearBasisValuesAtDofCoords(i,k,j);
-                }
-            }
+              li::getDofCoordsAndCoeffs(dofCoordsOriented,  dofCoeffsPhys, basisPtr, elemOrts);
 
-            Fun fun;
+              //Compute physical Dof Coordinates
+              {
+                Basis_HGRAD_TET_C1_FEM<DeviceSpaceType,ValueType,ValueType> tetLinearBasis; //used for computing physical coordinates
+                DynRankView ConstructWithLabel(tetLinearBasisValuesAtDofCoords, numCells, tet.getNodeCount(), basisCardinality);
+                for(ordinal_type i=0; i<numCells; ++i)
+                  for(ordinal_type d=0; d<dim; ++d) {
+                    auto inView = Kokkos::subview( dofCoordsOriented,i,Kokkos::ALL(),Kokkos::ALL());
+                    auto outView =Kokkos::subview( tetLinearBasisValuesAtDofCoords,i,Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL());
+                    tetLinearBasis.getValues(outView, inView);
+                    DeviceSpaceType().fence();
+                    for(ordinal_type j=0; j<basisCardinality; ++j)
+                      for(std::size_t k=0; k<tet.getNodeCount(); ++k)
+                        physDofCoords(i,j,d) += vertices[tets[i][k]][d]*tetLinearBasisValuesAtDofCoords(i,k,j);
```

Comment:
Could switch this to a parallel_for (you'd need to make `vertices` a Kokkos::Array to get transparent lambda capture).